libdnf/dnf-context.cpp

```diff
@@ -2271,16 +2269,43 @@ dnf_context_setup(DnfContext *context,
 
     /* setup a file monitor on the rpmdb, if we're operating on the native / */
     if (g_strcmp0(priv->install_root, "/") == 0) {
-        rpmdb_path = g_build_filename(priv->install_root, "var/lib/rpm/Packages", NULL);
-        file_rpmdb = g_file_new_for_path(rpmdb_path);
-        priv->monitor_rpmdb = g_file_monitor_file(file_rpmdb,
-                               G_FILE_MONITOR_NONE,
-                               NULL,
-                               error);
-        if (priv->monitor_rpmdb == NULL)
-            return FALSE;
-        g_signal_connect(priv->monitor_rpmdb, "changed",
-                         G_CALLBACK(dnf_context_rpmdb_changed_cb), context);
+        g_autofree char *rpmdb_path = rpmGetPath("%{_dbpath}", NULL);
+        g_autofree char *rpmdb_backend = rpmExpand("%{?_db_backend}", NULL);
+
+        if (rpmdb_path && rpmdb_backend) {
+            /* List of known RPM backends with their db files names; keep in sync
+               with RPM project's lib/backend/dbi.c:backends[] and its members.
+               or add public API to get to the actual DB file name into RPM */
+            struct _rpm_backends {
+                const char *name;
+                const char *filename;
+            } rpm_backends[] = {
+                { "sqlite", "rpmdb.sqlite" },
+                { "bdb_ro", "Packages" },
```

Comment:
Also note `bdb` too, since older versions of RPM have a rw `bdb` backend too.

Reply:
That is not part of the rpm 4.18.0. No idea which version made it read-only material though. You know, this is just the example why this map is not ideal.

Reply:
The backend was dropped in 4.18.0. It exists in 4.17.0 and older.